apps/mull-api/src/app/event/event.service.ts

```diff
@@ -47,33 +46,40 @@ export class EventService {
       .getMany();
   }
 
-  /**
-   * Find all the events that the user has not joined, created or is not co-hosting
-   * @param userId
-   */
-  findDiscoverEvent(userId: number): Promise<Event[]> {
+  findJoinedEvent(eventId: number, userId: number): Promise<Event> {
     return this.eventRepository
       .createQueryBuilder('event')
+      .leftJoin('event.participants', 'user')
       .leftJoin('event.coHosts', 'coHost')
-      .leftJoin('event.participants', 'participant')
       .leftJoin('event.host', 'host')
       .where(
         new Brackets((qb) => {
-          qb.where('coHost.id != :userId', { userId }).orWhere('coHost.id IS NULL');
-        })
-      )
-      .andWhere(
-        new Brackets((qb) => {
-          qb.where('participant.id != :userId', { userId }).orWhere('participant.id IS NULL');
+          qb.where('coHost.id = :userId', { userId })
+            .orWhere('user.id = :userId', { userId })
+            .orWhere('host.id = :userId', { userId });
         })
       )
       .andWhere(
         new Brackets((qb) => {
-          qb.where('host.id != :userId', { userId }).orWhere('host.id IS NULL');
+          qb.where('event.id = :eventId', { eventId });
         })
       )
-      .andWhere('event.restriction = :restriction', { restriction: String(EventRestriction.NONE) })
-      .getMany();
+      .getOne();
+  }
+
+  /**
+   * Find all the events that the user has not joined, created or is not co-hosting
+   * @param userId
+   */
+  async findDiscoverEvent(userId: number): Promise<Event[]> {
+    const discoverableEvents = await this.eventRepository.query(
+      `
+    SELECT DISTINCT id, title, startDate, endDate, description, imageId, hostId, locationId, restriction FROM event LEFT JOIN event_participants ON event.id = event_participants.eventId
+    WHERE id NOT IN (SELECT event.id FROM event WHERE event.hostId = ${userId}) AND 
+    id NOT IN (SELECT event_participants.eventId FROM event_participants WHERE event_participants.userId = ${userId}) AND 
+    id NOT IN (SELECT event_cohosts.eventId FROM event_cohosts WHERE event_cohosts.userId = ${userId})`
```

Comment:
This is just for curiosity to know if this way is better

Reply:
Yeah writing raw SQL isn't very appealing imo. What was wrong with the previous approach?

Reply:
In the previous approach, the discovered events data wouldn't be displayed correctly if there's more than one person that joined an event, there's most likely a way to do subqueries using query builder, but I'm not too familiar with that.

Reply:
This could probably be translated to the ORM syntax would just take some extra work. 
It was a complicated query which involved a lot of thinking, testing and debugging in workbench using SQL, so it was easier to just bring over that working code.

Reply:
Oh yeah that's fine then, we can create a new ticket possibly to address this later.